docs/io/fits/index.rst

```diff
@@ -178,6 +178,22 @@ When writing a file (e.g., with the :func:`writeto` function), compression will
 be determined based on the filename extension given, or the compression used in
 a pre-existing file that is being written to.
 
+
+Working with non-standard files
+"""""""""""""""""""""""""""""""
+`astropy.io.fits` adheres to the FITS standard and verifies that all fields within a FITS file conform
+to the standard. It may raise ``VerifyError`` if a field is deemed to be non-compliant. This action can be modified
+using method ``HDUList.verify``. It is typically invoked after opening the file but before reading any headers or data::
```

Comment:
Suggestion:

When astropy reads a FITS file which does not conform to FITS standard, it will not raise an error and exit. It will try to make the best educated interpretation and only gives up when the offending data is accessed and no unambiguous interpretation can be reached. If the file contains some invalid keyword, it may trigger warnings later when printing the header, or an exception when trying to write the file.
The file can be verified and fixed with `HDUList.verify`, and this can also be controlled when writing the file with the `output_verify` parameter.

Reply:
I am very happy with this overall direction as it looks like we are converging on a sensible addition to the FITS documentation.

I am confused in the first sentence when when you say "it will not raise an error and exit". This is the exact opposite of my experience: astropy does raise an error and exit when reading  non-compliant FITS files generated by the solar data observatory HMI instrument, (unless you include the "verify" option).  This is the reason why I decided to write the extra paragraph in the FITS documentation, as it took me some time and false attempts to find the `verify` solution, e.g I tried using `output_verify` which is documented in the `open` method but that did not work for me when reading files.

It seems from the previous messages that you are basing your statement _"it will not raise an error and exit"_ upon information on the verification page (http://docs.astropy.org/en/latest/io/fits/usage/verification.html). I believe this sentence is slightly misleading. I think it should say _when astropy reads a FITS file which does not conform to FITS standard, it will not IMMEDIATELY raise an error and exit_. 

The current version of Astropy does eventually raise an error if it cannot decode a value when reading a file. For example, I raise an error near line 766 of card.py in method `_parse_value` when I try to read my FITS file. If I look at this code I see,

```
        m = self._value_NFSC_RE.match(self._split()[1])

        if m is None:
            raise VerifyError("Unparsable card ({}), fix it first with "
                              ".verify('fix').".format(self.keyword))

```
Clearly a `VerifyError` is raised if `m is None` when `self._split()[1]` fails to match the regular expression. 

Am I making some mistake here and just missing the point? I apologize if that is the case.
I am new to making modifications to Astropy and I am still feeling my way through the system.

I think if we can get this issue clarified the pull request will sort itself out quite quickly. I am happy to run some test cases if that would help clarify any questions.

Reply:
OK. I think I am slowly coming up to speed. Are you suggesting that we have a paragraph in the documentation that is consistent with the changes applied by @dhomeier  with #10171 ? This would then make your suggested paragraph make sense to me.

Reply:
#10171 will fix the crash with your file when reading the data. So then the sentence "it will not raise an error and exit" will be true (hopefully, there may still be cases where it could crash, if a required keyword is invalid, but hopefully this should not be so frequent).